pkg/operator/ceph/file/filesystem.go

```diff
@@ -108,6 +108,13 @@ func deleteFilesystem(
 		logger.Warningf("continuing to remove filesystem CR even though downing the filesystem failed. %v", err)
 	}
 
+	// TODO: should we move the `RemoveFilesystem()` call to be before removing MDSes? If the below
+	// fails because the FS isn't empty, is it better to leave the filesystem active in case admins
+	// want to recover data from it?
+	//
+	// Additionally, if PreserveFilesystemOnDelete is set, we won't have Ceph's safety net to do one
+	// last check to see if the FS is in use before we delete it.
```

Comment:
@leseb @travisn ?

Reply:
Are you saying the filesystem is removed before the check for existing subvolumes? If it's already been confirmed that there are no subvolumes, seems like it doesn't matter if the mds or filesystem are removed first. Either way, it will be left in an unusable state, right? But if we haven't confirmed if subvolumes exist, seems like that should be first.

Reply:
I mean that there is a corner case where Rook removes MDSes while the filesystem is in-use. If we move the `RemoveFilesystem()` call to before removing deployments, Ceph will error out on fs removal, and we will still leave MDSes up for users to gracefully get their data off if necessary.

Reply:
If the filesystem deletion will error out when it is in use, can we just use that instead of adding all the subvolume checking code? In any case, I can't think why we shouldn't change that ordering.

Reply:
To my understanding, it errors only if there are connected clients. It does not error if there are existing subvolumes, otherwise it would already be blocking without these changes.